### ISSUE TITLE
network timeout on response to llama.cpp there is a hardcoded/undocumented network timeout 5m

### ISSUE THREAD

**Issue by aliaj00** [cline/cline#9328]
### Plugin Type

CLI

### Cline Version

2.2.1

### What happened?

while working on llama.cpp on long context work for my machine 20K and higher the Cline send cancel task:

srv  params_from_: Chat format: Qwen3 Coder
slot get_availabl: id  0 | task -1 | selected slot by LCP similarity, sim_best = 0.966 (> 0.100 thold), f_keep = 1.000
slot launch_slot_: id  0 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist 
slot launch_slot_: id  0 | task 13565 | processing task, is_child = 0
slot update_slots: id  0 | task 13565 | new prompt, n_ctx_slot = 262144, n_keep = 0, task.n_tokens = 50843
slot update_slots: id  0 | task 13565 | n_past = 49113, slot.prompt.tokens.size() = 49114, seq_id = 0, pos_min = 49113, n_swa = 1
slot update_slots: id  0 | task 13565 | restored context checkpoint (pos_min = 47386, pos_max = 47386, size = 75.376 MiB)
slot update_slots: id  0 | task 13565 | n_tokens = 47387, memory_seq_rm [47387, end)
slot update_slots: id  0 | task 13565 | prompt processing progress, n_tokens = 48411, batch.n_tokens = 1024, progress = 0.952166
slot update_slots: id  0 | task 13565 | n_tokens = 48411, memory_seq_rm [48411, end)
slot update_slots: id  0 | task 13565 | prompt processing progress, n_tokens = 49435, batch.n_tokens = 1024, progress = 0.972307
srv          stop: cancel task, id_task = 13565

i have been trying to find out where to increase the timeout like the option on the VSCODE plugin but could not find it for the cline cli version

but there seem to be a setting one the file "https://github.com/cline/cline/blob/34f07952173d23399b385606ee9907dbcc944a20/proto/cline/models.proto#L217"

there is  line 217 which says "optional int64 request_timeout_ms = 27;"

if i put it on the config files for the openai config it removes it  globalsettings.json and the text base menu does not have it as an option or on the docs online. 

Please can you help with it.

=Tim 

### Steps to reproduce

1.
2.
3.

**Comment by aliaj00** [comment]
steps to reproduce any very long prompt that goes more then 5 min to process from llama.cpp running qwen3-next 2bit from unsloth and watch the prompt on the server.
provider/model Qwen3-Coder-Next-UD-IQ2_XXS

system information: debian 12 nodejs 20.20.0

**Comment by ClineXDiego** [comment]
Thanks for the detailed investigation! You're right that requestTimeoutMs exists in the proto definition but isn't currently exposed in the CLI settings. The -t/--timeout flag controls the overall yolo mode task timeout (default 600s), not the per-API-request timeout.

For a workaround, you could try using the Ollama provider instead of OpenAI-compatible, since the Ollama handler does respect requestTimeoutMs. But you're correct that this setting should be surfaceable in the CLI for all providers — we'll track this as an enhancement.

**Comment by John-Ridgway** [comment]
I'm having this issue in the latest Cline VS Code extension with Ollama specified as the provider. The context is large and takes a long time to process, at 5 minutes exactly, it gets cancelled/timeout. I have the parameter in the provider panel specified as 999999999 ms, but I still get the 5m timeout as if it was a default fetch call.

EDIT:

<img width="668" height="611" alt="Image" src="https://github.com/user-attachments/assets/f38f351f-7b7e-4a8c-830c-882608670819" />

Log for reference:
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.729-04:00 level=INFO source=server.go:259 msg="enabling flash attention"
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.729-04:00 level=INFO source=server.go:433 msg="starting runner" cmd="/usr/local/bin/ollama runner --ollama-engine --model /usr/share/ollama/.ollama/models/blobs/sha256-83c54730a5fea8a0958598c01617c1419c431e93b33bacf980b49a420c798926 --port 44719"
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.730-04:00 level=INFO source=sched.go:484 msg="system memory" total="49.0 GiB" free="45.9 GiB" free_swap="13.0 GiB"
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.730-04:00 level=INFO source=server.go:792 msg="loading model" "model layers"=65 requested=-1
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.753-04:00 level=INFO source=runner.go:1450 msg="starting ollama engine"
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.770-04:00 level=INFO source=runner.go:1485 msg="Server listening on 127.0.0.1:44719"
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.774-04:00 level=INFO source=runner.go:1297 msg=load request="{Operation:fit LoraPath:[] Parallel:1 BatchSize:512 FlashAttention:Enabled KvSize:32000 KvCacheType: NumThreads:7 GPULayers:[] MultiUserCache:false ProjectorPath: MainGPU:0 UseMmap:false}"
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.821-04:00 level=INFO source=ggml.go:144 msg="" architecture=qwen35 file_type=Q4_K_M name="" description="" num_tensors=1307 num_key_values=53
May 06 17:05:31 LW11DS649 ollama[203]: load_backend: loaded CPU backend from /usr/local/lib/ollama/libggml-cpu-alderlake.so
May 06 17:05:31 LW11DS649 ollama[203]: time=2026-05-06T17:05:31.825-04:00 level=INFO source=ggml.go:104 msg=system CPU.0.SSE3=1 CPU.0.SSSE3=1 CPU.0.AVX=1 CPU.0.AVX_VNNI=1 CPU.0.AVX2=1 CPU.0.F16C=1 CPU.0.FMA=1 CPU.0.BMI2=1 CPU.0.LLAMAFILE=1 CPU.1.LLAMAFILE=1 compiler=cgo(gcc)
May 06 17:05:32 LW11DS649 ollama[203]: time=2026-05-06T17:05:32.348-04:00 level=INFO source=runner.go:1297 msg=load request="{Operation:alloc LoraPath:[] Parallel:1 BatchSize:512 FlashAttention:Enabled KvSize:32000 KvCacheType: NumThreads:7 GPULayers:[] MultiUserCache:false ProjectorPath: MainGPU:0 UseMmap:false}"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.938-04:00 level=INFO source=runner.go:1297 msg=load request="{Operation:commit LoraPath:[] Parallel:1 BatchSize:512 FlashAttention:Enabled KvSize:32000 KvCacheType: NumThreads:7 GPULayers:[] MultiUserCache:false ProjectorPath: MainGPU:0 UseMmap:false}"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.938-04:00 level=INFO source=ggml.go:490 msg="offloading 0 repeating layers to GPU"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.938-04:00 level=INFO source=ggml.go:494 msg="offloading output layer to CPU"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.938-04:00 level=INFO source=ggml.go:502 msg="offloaded 0/65 layers to GPU"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.939-04:00 level=INFO source=device.go:245 msg="model weights" device=CPU size="16.2 GiB"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.939-04:00 level=INFO source=device.go:256 msg="kv cache" device=CPU size="5.6 GiB"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.939-04:00 level=INFO source=device.go:267 msg="compute graph" device=CPU size="811.3 MiB"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.939-04:00 level=INFO source=device.go:272 msg="total memory" size="22.6 GiB"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.939-04:00 level=INFO source=sched.go:561 msg="loaded runners" count=2
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.939-04:00 level=INFO source=server.go:1385 msg="waiting for llama runner to start responding"
May 06 17:05:34 LW11DS649 ollama[203]: time=2026-05-06T17:05:34.940-04:00 level=INFO source=server.go:1428 msg="waiting for server to become available" status="llm server loading model"
May 06 17:05:43 LW11DS649 ollama[203]: time=2026-05-06T17:05:43.805-04:00 level=INFO source=server.go:1432 msg="llama runner started in 12.08 seconds"
May 06 17:10:58 LW11DS649 ollama[203]: [GIN] 2026/05/06 - 17:10:58 | 500 |          5m0s |       127.0.0.1 | POST     "/api/chat"

**Comment by bricss** [comment]
This network timeout ⏲️ is absolute cinema 🎥 and should be removed completely 🪓

**Comment by John-Ridgway** [comment]
> This network timeout ⏲️ is absolute cinema 🎥 and should be removed completely 🪓

I actually took hadsie:ollama-undici-headers-timeout https://github.com/cline/cline/pull/6554 and incorporated it into the main repo to locally build the extension. Seems to be working so far. I'll just roll with a custom version until someone decides it's worth bringing to the main.

**Comment by slavap** [comment]
I have 5m timeout problem in VSCode Cline extension, despite of having "requestTimeoutMs": 1800000 in globalState.json
Provider is "OpenAI Compatible", i.e. local llama.cpp
I think the following issue in RooCode describes this problem somehow: https://github.com/RooCodeInc/Roo-Code/issues/11932

**Comment by adminha** [comment]
I too have this problem. request times out on local running server (openai compatible) and I'm unable to define a timeout myself

**Comment by bricss** [comment]
I lost hope 🤞 waiting ⏳ for this issue to be fixed and completely switched 🔀 to a different coding agent 🤖 that works like a charm ✨ so far 🤌

**Comment by gunnar-schulze** [comment]
I experience the same issue. Would love to see a fix. Best regards.
